app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1292,6 +1294,42 @@ export default Controller.extend(bufferedProperty("model"), {
     this.model.destroy(this.currentUser);
   },
 
+  retryOnRateLimit(times, promise, topicId) {
+    const currentTopicId = this.get("model.id");
+    topicId = topicId || currentTopicId;
+    if (topicId !== currentTopicId) {
+      // we navigated to another topic, so skip
+      return;
+    }
+
+    if (this.retryRateLimited || times <= 0) {
```

Comment:
This variable is never initialized. We should add `retryRateLimited: null` at the beginning of the class. I would go further and suggest `this._retryRateLimited` because it's never observed and is private.